src/__main__.py

```diff
@@ -55,6 +62,40 @@ def quit_server():
     server.stop()
     return 'good-bye'
 
+def allowed_file(filename):
+    return '.' in filename and \
+           filename.rsplit('.', 1)[1].lower() in ALLOWED_EXTENSIONS
+
+@app.route('/', methods=['GET', 'POST'])
+def upload_file():
+    if request.method == 'POST':
+        # check if the post request has the file part
+        if 'file' not in request.files:
+            flash('No file part')
+            return redirect(request.url)
+        file = request.files['file']
+        # if user does not select file, browser also
+        # submit an empty part without filename
+        if file.filename == '':
+            flash('No selected file')
+            return redirect(request.url)
+        if file and allowed_file(file.filename):
+            filename = secure_filename(file.filename)
+            file.save(os.path.join(app.config['UPLOAD_FOLDER'], filename))
+            return render_template('index.html')
+    return '''
```

Comment:
So if I understand this correctly, when the user uploads a supported file extension, the page reloads, and when the user uploads an file with an unsupported file extension, the page switches to this other page, and stays there until the user uploads a valid file?

Reply:
yup, i just changed it so that if the file isnt supported it also refreshes. we can change this in the future to give a message when a file is successfully uploaded or not, though im not sure how to use python to make a webpage dynamic yet